src/commands/getProxyUrl.ts

```diff
@@ -0,0 +1,24 @@
+import * as vscode from 'vscode';
+import { logger, config } from '../extension';
+import { getDebugConfigFolder } from '../utility';
+import { isTokenExpired } from '../validateCredentials';
+import { launchDebugProxyCommandName } from '.';
+
+export async function getProxyUrl(cfg?: vscode.DebugConfiguration & { rootPath?: string; }) {
+  try {
+    const folder = getDebugConfigFolder(cfg);
+    const credentials = await config.getStoredCloudCredentials();
+    if (!credentials || isTokenExpired(credentials.token)) { return; }
```

Comment:
Prompt an error if the proxy is not started because of invalid credentials here?